Bypass SOPS by adding dry run mode

maorohana-redis opened the pull request:
This PR is providing an option to bypass SOPS runs when verifying only the kustomize overlay in CI workflows, enabled via an environment variable (SOPS_DRY_RUN).

maorohana-redis committed:
Add dry_run mode

jcassee commented:
Hi @maorohana-redis, thank you for the PR. I see the value in being able to verify the structure of the kustomization result without access to the SOPS keys.

A few questions:
* Is `SOPS_DRY_RUN` the best variable name? I'm not sure we are dry-running sops as such.
* Is leaving the sops-encrypted data the right approach? A static string like "sops-encrypted data omitted" might be clearer.
* You only touched the decryption of files. Could you also handle envs?
* Could you write some tests for this functionality?
* Could you add documentation to the README?

maorohana-redis committed:
SOPS dry run

maorohana-redis committed:
Add readme

maorohana-redis commented:
Hi @jcassee thanks for your fast response:

1. I'm not sure it's the best name , maybe "SKIP_SOPS_DECRYPTION"?
2. Now basically when we parse envs or file sources i set each key value to a placeholder value.
3. Added
4. Added test for handling envs + files.
5. Added

maorohana-redis committed:
Use SOPS_ENCRYPTED_DATA_OMITTED

maorohana-redis committed:
Change README

maorohana-redis committed:
type

maorohana-redis committed:
missing tests

maorohana-redis commented:
@jcassee can you review ?

maorohana-redis commented:
@jcassee waiting for your input
